tests/test_context_managers.py

```diff
@@ -412,6 +420,103 @@ def test_query_counter_ignores_particular_queries(self):
             )  # queries on db.system.indexes are ignored as well
             assert q == 1
 
+    def test_updating_a_document_within_a_transaction(self):
+        connect("mongoenginetest")
+        mongodb_version = get_mongodb_version()
+
+        class A(Document):
+            name = StringField()
+
+        A.drop_collection()
+
+        a_doc = A.objects.create(name="a")
+
+        if mongodb_version[0] >= 4 and IS_PYMONGO_GTE_37:
```

Comment:
Use something similar to @requires_mongodb_gte_44 , checking for mongodb version should be sufficient as pymongo  < 3.7 shouldn't be used with mongodb 4 (https://docs.mongodb.com/drivers/pymongo/#mongodb-compatibility), there was a bug in our CI related with a wrong mapping but it was fixed recently

Reply:
I've modified the tests to use the recommended approach of decorating for skipping.